### ISSUE TITLE
Implement the interaction with the prekey server

### ISSUE THREAD

**Issue by claucece** [twstrike/libotrv4#42]
As it will be defined in [issue 77](https://github.com/otrv4/otrv4/issues/77). This will replace the mock of #36 , but keep in mind what has been implemented there.

**Comment by claucece** [comment]
As it would probably be defined here: https://github.com/otrv4/otrv4-prekey-server

**Comment by claucece** [comment]
This includes (maybe): 
- [x] Structure of prekey server.
- [x] Validation of each prekey message returned by server.
- [x] DAKEZ with server: when does it start?
- [x] Query/or similar to server.
- [x] Incorporate a sending data message on non interactive msg from client. 
- [x] Questions: 
  - [x] can the enc msg on non interactive auth append tlvs?  No
  - [x] can the enc msg on non interactive auth have flags?  No
  - [x] will the receive of enc msg advertise a new ratchet? Yes

**Comment by claucece** [comment]
For the moment, please, don't look at this issue. It should be moved once we move the work to the prekey server spec is done.

**Comment by claucece** [comment]
This is duplicate of #98 . I'm closing this.
